eng/native/configureplatform.cmake

```
@@ -500,3 +500,9 @@ if(LOWERCASE_CMAKE_BUILD_TYPE STREQUAL debug)
     string(REPLACE "-D_FORTIFY_SOURCE=2 " "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
     string(REPLACE "-D_FORTIFY_SOURCE=2 " "" CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
 endif()
+
+if (CLR_CMAKE_TARGET_ANDROID OR CLR_CMAKE_TARGET_MACCATALYST OR CLR_CMAKE_TARGET_IOS OR CLR_CMAKE_TARGET_TVOS OR CLR_CMAKE_HOST_ARCH_WASM OR CLR_CMAKE_HOST_ARCH_ARMV6)
+    # Our zlib-ng copy should only be used in non-mobile and non-browser platforms only.
+    # In mobile platforms we should use the zlib package provided by the system.
```

```suggestion
    # Some platforms are opted-out from using the in-tree zlib-ng by default:
    # Android and iOS-like platforms: concerns about extra binary size
    # Wasm: zlib-ng breaks the build
    # Armv6: zlib-ng has build breaks
```
This comment is not 100% correct. Wasm is not the same as browser, and Armv6 is not really a mobile platform. I do not think that comments that try to rephrase the code are useful. Good comments explain why it is written like that.

---

Makes sense, thanks. I'll apply this suggestion when I get a proper fix for the wasm failure to avoid resetting the CI unnecessarily.